### PR TITLE
[NEW] API to logout from other locations

### LINKED ISSUE #16168: New API route to logout from all other sessions
In the UI, there is a button 'Logout from other logged in locations'. It would be great to have an API for that as well!

### PR DESCRIPTION
**Pull request by jschirrmacher**
Closes #16168 

Like discussed with @geekgonecrazy, this PR adds a new API route `/users.logoutOtherClients` so that this functionality which already exists in the UI (user profile) is available for automation as well.

### CODE REVIEW

File: tests/end-to-end/api/01-users.js [hunk 1]
```diff
@@ -16,6 +16,47 @@ import { customFieldText, clearCustomFields, setCustomFields } from '../../data/
 import { updatePermission, updateSetting } from '../../data/permissions.helper';
 import { createUser, login } from '../../data/users.helper.js';
 
+function createTestUser() {
```
Comment (MarcosSpessatto) [inline tests/end-to-end/api/01-users.js]: We already have a function to create tests users as you can see [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/tests/data/users.helper.js#L4).

File: tests/end-to-end/api/01-users.js [hunk 2]
```diff
@@ -16,6 +16,47 @@ import { customFieldText, clearCustomFields, setCustomFields } from '../../data/
...
+		const username = `user.test.${ Date.now() }`;
+		const email = `${ username }@rocket.chat`;
+		request.post(api('users.create'))
+			.set(credentials)
+			.send({ email, name: username, username, password })
+			.end((err, res) => resolve(res.body.user));
+	});
+}
+
+function loginTestUser(user) {
```
Comment (MarcosSpessatto) [inline tests/end-to-end/api/01-users.js]: Same as above here. We have a login helper [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/tests/data/users.helper.js#L13).

File: tests/end-to-end/api/01-users.js [hunk 3]
```diff
@@ -16,6 +16,47 @@ import { customFieldText, clearCustomFields, setCustomFields } from '../../data/
...
+				const userCredentials = {};
+				userCredentials['X-Auth-Token'] = res.body.data.authToken;
+				userCredentials['X-User-Id'] = res.body.data.userId;
+				resolve(userCredentials);
+			})
+			.end((err) => (err ? reject(err) : resolve()));
+	});
+}
+
+function deleteTestUser(user) {
```
Comment (MarcosSpessatto) [inline tests/end-to-end/api/01-users.js]: I would be nice if you can move this function to our users helpers [file](https://github.com/RocketChat/Rocket.Chat/blob/develop/tests/data/users.helper.js). :grimacing:

File: tests/end-to-end/api/01-users.js [hunk 4]
```diff
@@ -1741,4 +1760,49 @@ describe('[Users]', function() {
...
+				const result = await request.get(api('me'))
+					.set(userCredentials);
+				return result.statusCode === 401;
+			}
+
+			async function tryAuthentication() {
+				if (await checkAuthenticationFails()) {
+					done();
+				} else if (++counter < 20) {
+					setTimeout(tryAuthentication, 1000);
```
Comment (MarcosSpessatto) [inline tests/end-to-end/api/01-users.js]: I wonder if this way the tests will not take too long to execute. :thinking:

Comment (jschirrmacher) [reply tests/end-to-end/api/01-users.js]: As stated in the code comments, the invalidation is done asychronously, so we need to wait. In the best case, it is only 1 second, but I have configured a timeout of 20 seconds as a maximum timeout to be sure that the test completes in a reasonable time.

File: tests/end-to-end/api/01-users.js (hunk 1 above)
Comment (jschirrmacher) [reply tests/end-to-end/api/01-users.js]: I use the helper code now

File: tests/end-to-end/api/01-users.js (hunk 2 above)
Comment (jschirrmacher) [reply tests/end-to-end/api/01-users.js]: Using helper code now here as well

File: tests/end-to-end/api/01-users.js (hunk 3 above)
Comment (jschirrmacher) [reply tests/end-to-end/api/01-users.js]: done

**Comment by MarcosSpessatto** [comment]
@jschirrmacher can you take a look on the broken tests?

**Comment by jschirrmacher** [comment]
Tests work now as expected

**Comment by mrsimpson** [comment]
@MarcosSpessatto Can you please re-evaluate the PR?

File: tests/end-to-end/api/01-users.js [hunk 5]
```diff
@@ -1758,4 +1736,49 @@ describe('[Users]', function() {
...
+				} else {
+					done('Session did not invalidate in time');
+				}
+			}
+
+			request.post(api('users.logoutOtherClients'))
```
Comment (rodrigok) [inline tests/end-to-end/api/01-users.js]:
```suggestion
			request.post(api('users.removeOtherTokens'))
```

File: tests/end-to-end/api/01-users.js [hunk 6]
```diff
@@ -419,41 +419,19 @@ describe('[Users]', function() {
...
 		});
 		before((done) => {
 			updatePermission('edit-other-user-info', ['admin', 'user']).then(done);
 		});
-		after((done) => {
-			request.post(api('users.delete')).set(credentials).send({
-				userId: user._id,
-			}).end(() => updatePermission('edit-other-user-info', ['admin']).then(done));
+		after(async () => {
+			await deleteUser(user);
```
[the remark this answers is not in the document]
Comment (jschirrmacher) [reply tests/end-to-end/api/01-users.js]: I obviously removed that part unintentionally.

**Pull request merged**